File: VanillaExpress/server/classes/SignupRequest.js
class SignupRequest {
    #username;
    #email;
    #password;
    #passwordConfirm;

    constructor({username, email, password, passwordConfirm}) {
        this.#username = username;
        this.#email = email;
        this.#password = password;
        this.#passwordConfirm = passwordConfirm;
    }

    get username() {
        return this.#username;
    }

    set username(value) {
        this.#username = value;
    }

    get email() {
        return this.#email;
    }

    set email(value) {
        this.#email = value;
    }

    get password() {
        return this.#password;
    }

    set password(value) {
        this.#password = value;
    }

    get passwordConfirm() {
        return this.#passwordConfirm;
    }

    set passwordConfirm(value) {
        this.#passwordConfirm = value;
    }
}

module.exports = SignupRequest;

File: ReactSpring/backend/src/main/java/com/todolist/backend/security/request/SignupRequest.java
package com.todolist.backend.security.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@NoArgsConstructor
public class SignupRequest {

    @NotBlank
    @Size(min = 5)
    private String username;

    @NotBlank
    @Email
    @Size(max = 50)
    private String email;

    @NotBlank
    @Size(min = 12)
    private String password;

    @NotBlank
    @Size(min = 12)
    private String passwordConfirm;
}
